commit d53ba03481f06b936ab6ae9590eafd8560ca438b (refs/changes/83/1783/2)
Author: Joshua M. Clulow <jmc@joyent.com>
Date:   2017-04-12T00:20:08+00:00 (2 years, 6 months ago)
    
    joyent/node-progbar#18 treat all of the BSDs as UNIX platforms

diff --git a/.gitignore b/.gitignore
index 8753f80..9db000f 100644
--- a/.gitignore
+++ b/.gitignore
@@ -1,2 +1,4 @@
 node_modules
 *.*.swp
+*-?-stamp
+*.tgz
diff --git a/.npmignore b/.npmignore
new file mode 100644
index 0000000..751a9f4
--- /dev/null
+++ b/.npmignore
@@ -0,0 +1,5 @@
+*.*.swp
+*-?-stamp
+Makefile
+tests/
+.jshintrc
diff --git a/LICENSE b/LICENSE
index c9ba29b..d1c1b3e 100644
--- a/LICENSE
+++ b/LICENSE
@@ -1,4 +1,4 @@
-Copyright 2013 Joyent, Inc.  All rights reserved.
+Copyright (c) 2017, Joyent, Inc.
 
 Permission is hereby granted, free of charge, to any person obtaining a copy
 of this software and associated documentation files (the "Software"), to deal
diff --git a/Makefile b/Makefile
index bc7b11e..0df7377 100644
--- a/Makefile
+++ b/Makefile
@@ -2,6 +2,11 @@
 .PHONY: all
 all:
 
+npm-0-stamp:
+	npm install
+	touch $@
+
 .PHONY: check
-check:
-	@jshint lib/progbar.js
+check: npm-0-stamp
+	@./node_modules/.bin/jshint lib/progbar.js
+
diff --git a/lib/progbar.js b/lib/progbar.js
index e44b8a1..1968787 100644
--- a/lib/progbar.js
+++ b/lib/progbar.js
@@ -1,5 +1,5 @@
 /*
- * Copyright 2015 Joyent, Inc.  All rights reserved.
+ * Copyright (c) 2017, Joyent, Inc.
  */
 
 var mod_readline = require('readline');
@@ -72,7 +72,8 @@ if (process.env.TERM) {
 /*
  * On UNIX platforms, we can generally just write to the TTY.
  */
-var UNIX_PLATFORMS = [ 'sunos', 'solaris', 'darwin', 'linux' ];
+var UNIX_PLATFORMS = [ 'sunos', 'solaris', 'darwin', 'linux', 'openbsd',
+    'netbsd', 'freebsd' ];
 var USE_READLINE = (UNIX_PLATFORMS.indexOf(process.platform) === -1);
 
 /*
diff --git a/package.json b/package.json
index 57f9ca8..c092332 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
 	"name": "progbar",
 	"author": "Joshua M. Clulow <jmc@joyent.com>",
-	"version": "1.1.1",
+	"version": "1.2.0",
 	"description": "terminal progress bar",
 	"main": "./lib/progbar.js",
 	"contributors": [
@@ -13,7 +13,7 @@
 
 	"repository": {
 		"type": "git",
-		"url": "git://github.com/jclulow/node-progbar.git"
+		"url": "git://github.com/joyent/node-progbar.git"
 	},
 
 	"dependencies": {
@@ -22,6 +22,10 @@
 		"readable-stream": "~1.0.27-1"
 	},
 
+	"devDependencies": {
+		"jshint": "^2.7.0"
+	},
+
 	"keywords": [
 		"progress",
 		"bar",
